daemon/main.go

```diff
@@ -127,6 +132,22 @@ func main() {
 	}
 }
 
+func setupPidFile() {
```

Comment:
Do we need to handle another agent still running? this would be a pidfile still in place that points to a running PID. We don't seem to handle another agent instance's socket already existing in initEnv() so maybe this isn't a situation we handle anyway.
At first glance, I would expect us to do a check similar to the one for `cilium cleanup` above and refuse to start the agent if another is already running.
@aanm do you know?

Reply:
Agreed

Reply:
Can this be done separately?

Reply:
Created a new issue for this https://github.com/cilium/cilium/issues/2061